### DISCUSSION TITLE
last_insert_id is None

### DISCUSSION THREAD

**Discussion by OpsRJ** [rbatis/rbatis#278] [General]
After execute rb.save("",&struct).await.unwrap();

the option last_insert_id is None as you can see in the picture bellow

![Screenshot from 2021-02-25 23-19-17](https://user-images.githubusercontent.com/1771890/109226977-e20e9f00-77bf-11eb-880d-a47a527a266b.png)

**Comment by zhuxiujia** [comment]
> After execute rb.save("",&struct).await.unwrap();
> 
> the option last_insert_id is None as you can see in the picture bellow
> 
> ![Screenshot from 2021-02-25 23-19-17](https://user-images.githubusercontent.com/1771890/109226977-e20e9f00-77bf-11eb-880d-a47a527a266b.png)

last_insert_id  only support ```mysql``` and ```sqlite```

postgres or mssql  will be return None

**Comment by OpsRJ** [comment]
So after save, how can I get the id or entire struct?

**Comment by zhuxiujia** [comment]
> So after save, how can I get the id or entire struct?

use 
```rust
let id = rbatis::plugin::snowflake::new_snowflake_id()
```
or 
```rust
let id = rbatis::plugin::object_id::ObjectId::default()
```
